### PR TITLE
Add custom dataset tutorial.

### PR DESCRIPTION
**Pull request by Gordon119**
## What does this PR do?

New dataset tutorial.
https://www.csie.ntu.edu.tw/~b09902131/html/auto_examples/plot_dataset_tutorial.html#

## Test CLI & API (`bash tests/autotest.sh`)
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_examples.sh`)
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
- [ ] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [x] Not Applicable (i.e., the PR does not include API changes.)

### CODE REVIEW

File: docs/examples/plot_dataset_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+"""
+An Example of Using Data Stored in Different Forms.
```
Comment (Eleven1Liu) [inline docs/examples/plot_dataset_tutorial.py:2]:
```suggestion
An Example of Using Data Stored in Different Forms
```

File: docs/examples/plot_dataset_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+data_sets['test'] = load_dataset('tweet_eval', 'emoji', split='test')
+
+######################################################################
+# Convert to LibMultiLabel format
+# --------------------------------------------------
+# We begin with loading Hugging Face's data set to Pandas' structure by the function ``DataFrame``. 
+# In consistence with our `linear model quickstart <https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_linear_quickstart.html>`_ which don't have validation set,
+# we use ``pandas.concat`` to merge training and validation sets and use ``reset_index`` to add the new indices to rows.
+
+for d_set in ['train', 'val', 'test']:
```
Comment (Eleven1Liu) [inline docs/examples/plot_dataset_tutorial.py:39]: How about replacing `d_set` with `split` which is aligned with [BERT's](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_bert_quickstart.html)?

Comment (Gordon119) [reply docs/examples/plot_dataset_tutorial.py]: Ok. Done.

File: docs/examples/plot_dataset_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+###############################################################################
+# The format of the data set after conversion looks like below::
+#
+#   >>> print(data_sets['train'].loc[[0]]) #print first row
+#   ...
+#   index  label                                               text
+#       0     12  Sunday afternoon walking through Venice in the...  
+#
+# Next, we train and make prediction with the data set. The detailed explanation is in our `linear model quickstart <https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_linear_quickstart.html>`_. 
+# The difference between here and the quickstart is that the ``data_format`` option should be ``'dataframe`` because the data set is a dataframe now.
```
Comment (Eleven1Liu) [inline docs/examples/plot_dataset_tutorial.py:54]: Remove `'` from ``'dataframe``?

Comment (Gordon119) [reply docs/examples/plot_dataset_tutorial.py]: Fixed.

File: docs/examples/plot_dataset_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+###############################################################################
+# The format of the data set after conversion looks like below::
+#
+#   >>> print(data_sets['train'].loc[[0]]) #print first row
+#   ...
+#   index  label                                               text
+#       0     12  Sunday afternoon walking through Venice in the...  
+#
+# Next, we train and make prediction with the data set using linear model. The detailed explanation is in our `linear model quickstart <https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_linear_quickstart.html>`_.
```
Comment (cjlin1) [inline docs/examples/plot_dataset_tutorial.py:53]:
```suggestion
# Next, we train and make prediction with the data set using a linear model. The detailed explanation is in our `linear model quickstart <https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_linear_quickstart.html>`_. 
```

File: docs/examples/plot_dataset_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+preprocessor = linear.Preprocessor(data_format='dataframe')
+datasets = preprocessor.load_data(data_sets['train'], data_sets['test'])
+
+###############################################################################
+# Also, if you want to use a NN model,
+# use ``load_datasets`` from ``datasets`` and change the data to the dataframes we created.
+# Here is the modification of our `Bert model quickstart <https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/auto_examples/plot_BERT_quickstart.html>`_. 
+
+from datasets import load_dataset
```
Comment (cjlin1) [inline docs/examples/plot_dataset_tutorial.py:66]: but the line below uses load_datasets???

Comment (Gordon119) [reply docs/examples/plot_dataset_tutorial.py]: Fixed.

File: docs/examples/plot_dataset_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# The difference between here and the quickstart is that the ``data_format`` option should be ``dataframe`` because the data set is a dataframe now.
+
+import libmultilabel.linear as linear
+
+preprocessor = linear.Preprocessor(data_format='dataframe')
+datasets = preprocessor.load_data(data_sets['train'], data_sets['test'])
+
+###############################################################################
+# Also, if you want to use a NN model,
+# use ``load_datasets`` from ``datasets`` and change the data to the dataframes we created.
```
Comment (cjlin1) [inline docs/examples/plot_dataset_tutorial.py:63]: incorrect.. not from "datasets"

**Pull request merged**